_source/_posts/2021-12-06-spring-data-elasticsearch.md

```diff
@@ -0,0 +1,349 @@
+jhipster jdl blog-reactive-ms.jdl
+```
+
+{% include setup/cli.md type="jhipster" %}
+
+Follow the process above for creating 2 applications in Okta, the gateway application, and the blog application.
+
+```shell
+cd gateway
+okta apps create jhipster
+```
+
+Make sure to set up http://localhost:8081/login/oauth2/code/oidc with port 8081 as Redirect URI for blog microservice.
+
+```shell
+cd blog
+okta apps create jhipster
+```
+
+Find the `blog` credentials in the files `blog/.okta.env` and  `gateway/.okta.env`.
+
+The JHipster Registry is also a Spring Cloud Config server, and by default, it is configured with the profiles `dev` and `native`, which means the configuration will be provided from the location `docker-compose/central-server/config`. Let's add some files so it can provide the OAuth2 configuration to the microservices.
+Add the file `docker-compose/central-server-config/blog-prod.yml` with the following content:
+
+```yml
+spring:
+  security:
+    oauth2:
+      client:
+        provider:
+          oidc:
+            issuer-uri: https://{yourOktaDomain}/oauth2/default
+        registration:
+          oidc:
+            client-id: {blogClientId}
+            client-secret: {blogClientSecret}
+```
+
+Add also the file `docker-compose/central-server-config/gateway-prod.yml` with the following content:
+
+```yml
+spring:
+  security:
+    oauth2:
+      client:
+        provider:
+          oidc:
+            issuer-uri: https://{yourOktaDomain}/oauth2/default
+        registration:
+          oidc:
+            client-id: {gatewayClientId}
+            client-secret: {gatewayClientSecret}
+```
+
+For the experimentation in this tutorial, configure the Kibana interface for Elasticsearch, which will allow visualizations of the Elasticsearch data.
+Edit `docker-compose/docker-compose.yml` and add the Kibana service like this:
+
+```yml
+services:
+  blog-kibana:
+     image: docker.elastic.co/kibana/kibana:7.13.3
+     ports:
+       - 5601:5601
+     environment:
+       ELASTICSEARCH_URL: http://blog-elasticsearch:9200
+       ELASTICSEARCH_HOSTS: '["http://blog-elasticsearch:9200"]'
+```
+
+Also add the following additional configuration to the elasticsearch service, to expose the ports to the local host for sending APIs requests:
+
+```yml
+services:
+  blog-elasticsearch:
+    ...
+    ports:
+      - 9200:9200
+```
+
+Create the applications container image:
+
+```shell
+cd blog
+./mvnw -DskipTests -ntp -Pprod verify jib:dockerBuild
+```
+
+```shell
+cd gateway
+./mvnw -DskipTests -ntp -Pprod verify jib:dockerBuild
+```
+
+Run the architecture with Docker Compose:
+
+```shell
+cd docker-compose
+docker compose up
+```
+ Access the JHipster Registry at http://localhost:8761 and sign in with user=admin, password=admin. When you see all services up and green, go to http://localhost:8080 and sign in with your Okta account.
```

Comment:
```suggestion

Access the JHipster Registry at `http://localhost:8761 `and sign in with `user=admin, password=admin`. When you see all services up and green, go to `http://localhost:8080` and sign in with your Okta credentials.
```

Reply:
admin/admin should be "your Okta credentials" if you configure Okta in the main `application.yml` for config server.

Reply:
Updated